examples/hal/quad/main.rs

```diff
@@ -612,8 +612,11 @@ fn main() {
     }
     for (image, rtv) in frame_images {
         device.destroy_image_view(rtv);
+        #[cfg(not(feature="vulkan"))]
```

Comment:
why aren't we doing that for Vulkan? We shouldn't need any config flags here

Reply:
Because in Vulkan when destorying the swapchain it also destroys images associated with the swapchain, if the images is destroyed before the swapchain the program crashes with a memory access violation

Reply:
We should just follow Vulkan then and not require the user to destroy them.

Reply:
You mean delete the line and config check? If so what about DX12 where the user should destroy the images?

Reply:
In the API we have (even if you don't consider Vulkan Portability initiative), since the user doesn't explicitly create swapchain images, they shouldn't be explicitly destroying them. DX12 needs to have some logic to handle this automatically, and complain loudly if the user tries to destroy those images manually.